src/robotFactory.js

```diff
@@ -1,10 +1,80 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  };
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  };
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  };
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  };
+
+  getInfo() {
+    let result = `Robot: ${this.name}, `;
+
+    result += `Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+
+    return result;
+  }
```

Comment:
Looks a little bit strange 

```suggestion
    return `Robot: ${this.name}, `
      + `Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
  }
```